Ing-Soft-2024/rescue-back
# feat: Add base categories to the database

- Add base categories to the database to provide a foundation for categorizing products.

refactor: Remove files and update commerce

- Remove unnecessary files and make changes to the commerce module.

feat: Add routes

- Add new routes to enhance the navigation and functionality of the application.

test: Add tests

- Include new tests to ensure the correctness and reliability of the code.

fix: Update port in docker-compose.yml

- Correct the port configuration in the docker-compose.yml file to resolve a port conflict.

feat: Implement login and registration functionality

- Implement the login and registration functionality in the credentials controller.

feat: Add getProductById function

- Add the getProductById function to retrieve a product by its ID.

fix: Update getProductById function

- Update the getProductById function to include additional functionality.

fix: Update moduleInterpreter function

- Update the moduleInterpreter function to handle cases where the module.default is a function.

fix: Update swagger configuration

- Update the swagger configuration to use the http scheme instead of https.

feat: Add addCommerce function

- Add the addCommerce function to create a new commerce entity in the database.

feat: Add getCategoryById function

- Add the getCategoryById function to retrieve a category by its ID.

feat: Add deleteProduct function

- Add the deleteProduct function to delete a product from the database.

feat: Add getCommerces function

- Add the getCommerces function to retrieve all commerce entities from the database.

feat: Add addProduct function

- Add the addProduct function to create a new product in the database.

fix: Update User model associations

- Update the associations in the User model to fix a bug.

feat: Add getListOfOrders function

- Add the getListOfOrders function to retrieve a list of orders based on user and business IDs.

feat: Add getCommerceById function

- Add the getCommerceById function to retrieve a commerce entity by its ID.

feat: Implement login functionality in credentials controller

- Implement the login functionality in the credentials controller.

fix: Update Order model

- Update the Order model to include the necessary attributes.

feat: Add editProduct function

- Add the editProduct function to update a product in the database.

feat: Add getListOfProducts function

- Add the getListOfProducts function to retrieve a list of products based on category and business IDs.

feat: Add CategoryDetails router

- Add the CategoryDetails router to manage category details.
